startLogcatBroadcast does not respect https

## Description

Bug.
The Appium is accessible over **HTTPS**.
```
curl https://<user>:<pass>@<host>:8443/status
{"value":{"ready":true,"message":"The server is ready to accept new connections","build":{"version":"2.1.3"}}}
```
The tests are triggered with java property -Dappium.hub=**https**://user:pass@host:8443.

When we call `startLogcatBroadcast`:
`androidDriver.startLogcatBroadcast(androidDriver.getRemoteAddress().getHost(), androidDriver.getRemoteAddress().getPort());
`
We get error:
```
org.openqa.selenium.remote.http.ConnectionFailedException: JdkWebSocket initial request execution error
Build info: version: '4.21.0', revision: '79ed462ef4'
System info: os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '14.5', java.version: '17.0.7'
Driver info: driver.version: unknown
Caused by: java.net.http.WebSocketHandshakeException
Caused by: jdk.internal.net.http.websocket.CheckFailedException: Unexpected HTTP response status code 404
	... 11 more
java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
org.openqa.selenium.remote.http.jdk.JdkHttpClient.openSocket(JdkHttpClient.java:240)
io.appium.java_client.ws.StringWebSocketClient.connect(StringWebSocketClient.java:74)
io.appium.java_client.android.ListensToLogcatMessages.startLogcatBroadcast(ListensToLogcatMessages.java:67)
```

URI: **http**://host:8443/ws/session/b515a27e-ff21-4e43-b8f2-b336eed55665/appium/device/logcat

[Here](https://github.com/appium/java-client/blob/e62b3951cb48bea076d104a3a3d0b6588e618587/src/main/java/io/appium/java_client/android/ListensToLogcatMessages.java#L62C13-L62C24) the endpointUri is set to `ws://` regardless  `-Dappium.hub=https://`. The `wss://` should be used.


## Environment

* Java client build version: java-client-9.2.3.jar
* Appium server version: 2.1.3
* OS version used to run Appium: Ubuntu 18.04.3 LTS
* Node.js version: v16.20.2
* Mobile platform/version under test: Android
* Real device: moto-e40 (Android 11)

--- a/src/main/java/io/appium/java_client/android/ListensToLogcatMessages.java
+++ b/src/main/java/io/appium/java_client/android/ListensToLogcatMessages.java
@@ -19,10 +19,12 @@ package io.appium.java_client.android;
 import io.appium.java_client.CommandExecutionHelper;
 import io.appium.java_client.ExecutesMethod;
 import io.appium.java_client.ws.StringWebSocketClient;
+import org.openqa.selenium.remote.HttpCommandExecutor;
 import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.remote.SessionId;
 
 import java.net.URI;
-import java.net.URISyntaxException;
+import java.net.URL;
 import java.util.function.Consumer;
 
 import static io.appium.java_client.service.local.AppiumServiceBuilder.DEFAULT_APPIUM_PORT;
@@ -36,7 +38,7 @@ public interface ListensToLogcatMessages extends ExecutesMethod {
      * is assigned to the default port (4723).
      */
     default void startLogcatBroadcast() {
-        startLogcatBroadcast("127.0.0.1", DEFAULT_APPIUM_PORT);
+        startLogcatBroadcast("127.0.0.1");
     }
 
     /**
@@ -56,15 +58,13 @@ public interface ListensToLogcatMessages extends ExecutesMethod {
      * @param port the port of the host where Appium server is running
      */
     default void startLogcatBroadcast(String host, int port) {
+        var remoteWebDriver = (RemoteWebDriver) this;
+        URL serverUrl = ((HttpCommandExecutor) remoteWebDriver.getCommandExecutor()).getAddressOfRemoteServer();
+        var scheme = "https".equals(serverUrl.getProtocol()) ? "wss" : "ws";
         CommandExecutionHelper.executeScript(this, "mobile: startLogsBroadcast");
-        final URI endpointUri;
-        try {
-            endpointUri = new URI(String.format("ws://%s:%s/ws/session/%s/appium/device/logcat",
-                    host, port, ((RemoteWebDriver) this).getSessionId()));
-        } catch (URISyntaxException e) {
-            throw new IllegalArgumentException(e);
-        }
-        getLogcatClient().connect(endpointUri);
+        SessionId sessionId = remoteWebDriver.getSessionId();
+        var endpoint = String.format("%s://%s:%s/ws/session/%s/appium/device/logcat", scheme, host, port, sessionId);
+        getLogcatClient().connect(URI.create(endpoint));
     }
 
     /**

--- a/src/main/java/io/appium/java_client/ios/ListensToSyslogMessages.java
+++ b/src/main/java/io/appium/java_client/ios/ListensToSyslogMessages.java
@@ -19,10 +19,12 @@ package io.appium.java_client.ios;
 import io.appium.java_client.CommandExecutionHelper;
 import io.appium.java_client.ExecutesMethod;
 import io.appium.java_client.ws.StringWebSocketClient;
+import org.openqa.selenium.remote.HttpCommandExecutor;
 import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.remote.SessionId;
 
 import java.net.URI;
-import java.net.URISyntaxException;
+import java.net.URL;
 import java.util.function.Consumer;
 
 import static io.appium.java_client.service.local.AppiumServiceBuilder.DEFAULT_APPIUM_PORT;
@@ -37,7 +39,7 @@ public interface ListensToSyslogMessages extends ExecutesMethod {
      * is assigned to the default port (4723).
      */
     default void startSyslogBroadcast() {
-        startSyslogBroadcast("localhost", DEFAULT_APPIUM_PORT);
+        startSyslogBroadcast("localhost");
     }
 
     /**
@@ -57,15 +59,13 @@ public interface ListensToSyslogMessages extends ExecutesMethod {
      * @param port the port of the host where Appium server is running
      */
     default void startSyslogBroadcast(String host, int port) {
+        var remoteWebDriver = (RemoteWebDriver) this;
+        URL serverUrl = ((HttpCommandExecutor) remoteWebDriver.getCommandExecutor()).getAddressOfRemoteServer();
+        var scheme = "https".equals(serverUrl.getProtocol()) ? "wss" : "ws";
         CommandExecutionHelper.executeScript(this, "mobile: startLogsBroadcast");
-        final URI endpointUri;
-        try {
-            endpointUri = new URI(String.format("ws://%s:%s/ws/session/%s/appium/device/syslog",
-                    host, port, ((RemoteWebDriver) this).getSessionId()));
-        } catch (URISyntaxException e) {
-            throw new IllegalArgumentException(e);
-        }
-        getSyslogClient().connect(endpointUri);
+        SessionId sessionId = remoteWebDriver.getSessionId();
+        var endpoint = String.format("%s://%s:%s/ws/session/%s/appium/device/syslog", scheme, host, port, sessionId);
+        getSyslogClient().connect(URI.create(endpoint));
     }
 
     /**

